File: students/students/src/MON.Models/RecognitionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace MON.Models
{
    public class RecognitionModel
    {
        public int? Id { get; set; }
        public int PersonId { get; set; }
        [MaxLength(1000)]
        public string InstitutionName { get; set; }
        public int? InstitutionCountryId { get; set; }
        public int EducationLevelId { get; set; }
        public int? SPPOOProfessionId { get; set; }
        public int? SPPOOSpecialityId { get; set; }
        public int? Term { get; set; }
        public int? BasicClassId { get; set; }
        [MaxLength(100)]
        public string RuoDocumentNumber { get; set; }
        public DateTime? RuoDocumentDate { get; set; }
        [MaxLength(100)]
        public string DiplomaNumber { get; set; }
        public DateTime? DiplomaDate { get; set; }
        public IEnumerable<DocumentModel> Documents { get; set; }
        public IEnumerable<RecognitionEqualizationModel> Equalizations { get; set; }
        public short? SchoolYear { get; set; }
        public string SchoolYearName { get; set; }
        public string SPPOOProfessionName { get; set; }
        public string SPPOOSpecialityName { get; set; }
        public string EducationLeveName { get; set; }
        public string BasicClassName { get; set; }
        public int? VetLevel { get; set; }
        public bool IsSelfEduForm { get; set; }
    }
}

File: students/students/src/MON.SPA/src/models/recognitionModel.js
import Constants from '@/common/constants.js';
import { DocumentModel } from './documentModel.js';

export class RecognitionModel {
  constructor(obj = {}, moment = {}) {
    this.id = obj.id || null;
    this.personId = obj.personId || null;
    this.institutionName = obj.institutionName || null;
    this.institutionCountryId = obj.institutionCountryId || null;
    this.educationLevelId = obj.educationLevelId || null;
    this.sppooProfessionId = obj.sppooProfessionId || null;
    this.sppooSpecialityId = obj.sppooSpecialityId || null;
    this.term = obj.term || null;
    this.basicClassId = obj.basicClassId || null;
    this.diplomaNumber = obj.diplomaNumber || null;
    this.diplomaDate = (obj.diplomaDate && moment) ? moment(obj.diplomaDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.ruoDocumentNumber = obj.ruoDocumentNumber || null;
    this.ruoDocumentDate = (obj.ruoDocumentDate && moment) ? moment(obj.ruoDocumentDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    this.equalizations = obj.equalizations || [];
    this.isSelfEduForm = obj.isSelfEduForm || false;

    // From the view model
    this.institutionCountry = obj.institutionCountry || '';
    this.basicClass = obj.basicClass || '';
    this.termName = obj.termName || '';
    this.educationLevel = obj.educationLevel || '';
    this.sppooProfession = obj.sppooProfession || '';
    this.sppooSpeciality = obj.sppooSpeciality || '';
    this.schoolYear = obj.schoolYear;
    this.schoolYearName = obj.schoolYearName;
    this.basicClassName = obj.basicClassName;
    this.educationLeveName = obj.educationLeveName;
    this.sppooProfessionName = obj.sppooProfessionName;
    this.sppooSpecialityName = obj.sppooSpecialityName;
    this.vetLevel = obj.vetLevel;
  }
}
